examples/Spirit/spirit_utils.cc

```diff
@@ -170,11 +176,222 @@ std::tuple<
+//                      minValUpper, 
+//                      maxValUpper  );
+                     
+setSpiritJointLimits( plant, 
+                     trajopt,  
+                     indicesLower, 
+                     minValLower, 
+                     maxValLower  );
+                     
+setSpiritJointLimits( plant, 
+                     trajopt,  
+                     indicesAbduc, 
+                     minValAbduc, 
+                     maxValAbduc  );
+
+}
+//   \SETSPIRITJOINTLIMITS 
+
+
+// **********************************************************
+//   SETSPIRITACTUATIONLIMITS
+template <typename T>  
+void setSpiritActuationLimits(drake::multibody::MultibodyPlant<T> & plant, 
+                          dairlib::systems::trajectory_optimization::Dircon<T>& trajopt,
+                          double actuatorLimit){
+  auto actuators_map = multibody::makeNameToActuatorsMap(plant);
+  // std::cout<<"joint_" + std::to_string(iJoint)<<std::endl;
+  int N_knotpoints = trajopt.N();
+  for(int iKnot = 0; iKnot<N_knotpoints;iKnot++){
+    auto ui = trajopt.input(iKnot);
+    for (int iMotor = 0; iMotor<12; iMotor++){
+      trajopt.AddBoundingBoxConstraint(-actuatorLimit,actuatorLimit,ui(actuators_map.at("motor_" + std::to_string(iMotor))));
+    }
+  }
+
+}
+//   \SETSPIRITACTUATIONLIMITS
+
+
+// **********************************************************
+//   SETSPIRITSYMMETRY
+
+template <typename T>
+void makeSaggitalSymmetric(
+        drake::multibody::MultibodyPlant<T> & plant, 
+        dairlib::systems::trajectory_optimization::Dircon<T>& trajopt){
+
+  // Get position and velocity dictionaries 
+  auto positions_map = multibody::makeNameToPositionsMap(plant);
+  auto velocities_map = multibody::makeNameToVelocitiesMap(plant);
+  int num_knotpoints = trajopt.N();
+  /// Symmetry constraints
+  for (int i = 0; i < num_knotpoints; i++){
+    
+    auto xi = trajopt.state(i);  
+    // Symmetry constraints (mirrored hips all else equal across)
+    trajopt.AddLinearConstraint( xi( positions_map.at("joint_8") ) == -xi( positions_map.at("joint_10") ) );
+    trajopt.AddLinearConstraint( xi( positions_map.at("joint_9") ) == -xi( positions_map.at("joint_11") ) );
+    // // Front legs equal and back legs equal constraints 
+    trajopt.AddLinearConstraint( xi( positions_map.at("joint_0") ) ==  xi( positions_map.at("joint_4") ) );
+    trajopt.AddLinearConstraint( xi( positions_map.at("joint_1") ) ==  xi( positions_map.at("joint_5") ) );
+    trajopt.AddLinearConstraint( xi( positions_map.at("joint_2") ) ==  xi( positions_map.at("joint_6") ) );
+    trajopt.AddLinearConstraint( xi( positions_map.at("joint_3") ) ==  xi( positions_map.at("joint_7") ) );
+  }
+}
+template <typename T>
+void setSpiritSymmetry(
+        drake::multibody::MultibodyPlant<T> & plant, 
+        dairlib::systems::trajectory_optimization::Dircon<T>& trajopt,
+        std::vector<std::string> symmetries){
+  
+  
+  static bool hasRun = false;
```

Comment:
Umm, yeah. Thats probably a silly way to do this since I will be calling this function multiple times in a valid way when running the multiple optimizations in one script..